pkg/scraper/bruneida_scraper.go

```diff
@@ -1,73 +1,127 @@
 package scraper
 
 import (
+	"errors"
 	"fmt"
-	"strconv"
+	"regexp"
 	"strings"
 
+	"github.com/PuerkitoBio/goquery"
 	"github.com/b-open/jobbuzz/pkg/model"
-	"github.com/gocolly/colly"
 )
 
-func ScrapeBruneida() []model.Job {
-	allowedDomain := colly.AllowedDomains("www.bruneida.com")
-	linkCollector := colly.NewCollector(
-		allowedDomain,
-	)
-	jobCollector := linkCollector.Clone()
-
+func ScrapeBruneida() ([]model.Job, error) {
 	jobs := []model.Job{}
 
-	// Scraping the links
-	linkCollector.OnHTML("ul.list-az.ul-azs", func(e *colly.HTMLElement) {
-		e.ForEach(".az-detail>h3.az-title>a.h-elips", func(i int, child *colly.HTMLElement) {
-			link := child.Attr("href")
-			jobCollector.Visit(link)
+	for i := 1; i < 2; i++ {
+		url := fmt.Sprintf("https://www.bruneida.com/brunei/jobs/?&page=%d", i)
 
-		})
-	})
+		links, err := getJobLinks(url)
 
-	// Scraping the jobs
-	jobCollector.OnHTML("body", func(h *colly.HTMLElement) {
-		job_title := h.ChildText("#title-box-inner div.inline-block.pull-left h1")
-		company := h.ChildText("#ad-contact ul li:first-child span.bb b.small")
-		salary := h.ChildText("#ad-body-inner .opt .opt-dl:nth-child(3) .dd")
+		if err != nil {
+			return nil, err
+		}
 
-		locations := []string{}
-		h.ForEach("#ad-body-inner .opt .opt-dl", func(i int, child *colly.HTMLElement) {
-			title := child.ChildText(".dt")
+		for _, link := range links {
+			job, err := scrapeBruneidaJob(link)
 
-			if strings.Contains(title, "City") || strings.Contains(title, "Local") {
-				locations = append(locations, child.ChildText(".dd"))
+			if err != nil {
+				continue
 			}
-		})
 
-		location := strings.Join(locations, " ")
+			jobs = append(jobs, *job)
```

Comment:
You can change `jobs` to be a `[]*model.Job` if you don't want to have to convert job to value just to append it.